0-base-protocol-background.md

```diff
@@ -0,0 +1,39 @@
+# IRC-0 Background on StateChannels Protocol
+
+## Overview
+State channels is a core protocol "engine" (they call it a `wallet`) that validates and signs channel updates.
+
+It is similar to our existing CFCore module in the following ways:
+1. The core "application" lifecycle of `propose`--> `install`--> `takeAction` --> `uninstall` is retained.
+2. Data flows through an external transport layer (for inbound messages from a counterparty) or JSON rpc interface (for user-initiated inputs), gets validated by the protocol against previous state, and then gets stored in store.
+3. The output of state transitions is a set of commitments that can be put onchain in the event of a dispute.
+
+It is different in the following ways:
+1. The "application" lifecycle is named `create`--> `join`--> `update`--> `close`. The same cycle is used both for "subchannels"/virtual channels (i.e. apps), **and** for the ledger channel. This dramatically simplifies the interfaces.
+2. The core protocol handles deposit/withdraw natively. //TODO How do channel topups and partial withdraws work?
+3. There are no more multisig contracts. Instead there's a global `assetHolder` contract per asset. //TODO How do we deposit directly to an address? How do we allow counterparty to pay for withdrawal?
+4. The StateChannels wallet does not handle message dispatching and responding directly. Instead, it uses a `pushMessage` pattern whereby the implementer pushes updates into the wallet. This dramatically simplifies adding middleware for encryption, message retries, caching, etc.
+5. The protocol extends turn based updating from being only an "app"-level feature to instead be throughout the protocol. This removes the need for distributed locks, dramatically reduces the risk of states getting out of sync with each other (TODO: verify), and allows parties to receive updates entirely out of order with no problems.
+6. The application solidity code no longer defines a state machine but instead a simpler interface which just ensures that a given transition is valid. This means that calculating state transitions themselves is outside the scope of the wallet.
+7. The wallet is more tightly coupled with the store implementation and cannot easily be abstracted into a general interface with pluggable stores.
```

Comment:
This is because wallets are making use of transactions within the store itself (i.e. database transactions in the `server-wallet`) to serialize channel operations (think button mashing). Not sure if this is the complete context for the new architecture, but it is an important one